python/lsst/pipe/supertask/pipelineBuilder.py

```diff
@@ -0,0 +1,285 @@
+#
+# LSST Data Management System
+# Copyright 2018 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+"""
+Module defining PipelineBuilder class and related methods.
+"""
+
+from __future__ import print_function
+from builtins import object
+
+__all__ = ['PipelineBuilder']
+
+#--------------------------------
+#  Imports of standard modules --
+#--------------------------------
+import os
+
+#-----------------------------
+# Imports for other modules --
+#-----------------------------
+from .configOverrides import ConfigOverrides
+from .pipeline import Pipeline, TaskDef
+from . import pipeTools
+import lsst.daf.persistence as dafPersist
+import lsst.log as lsstLog
+import lsst.utils
+
+#----------------------------------
+# Local non-exported definitions --
+#----------------------------------
+
+#------------------------
+# Exported definitions --
+#------------------------
+
+
+class PipelineBuilder(object):
+    """
+    PipelineBuilder class is responsible for building task pipeline from
+    a set of command line arguments.
```

Comment:
Most of functionality is already available but without specialized methods. You can modify Pipeline because it is essentially a list, and I have added just one extra method to it to make search by label easier. One non-trivial thing in PipelineBuilder that is worth exposing is a construction of TaskDef - importing task class and applying camera overrides - but that should not be a part of Pipeline class because it has non-trivial dependency. We'll need to think how to do it right as we go forward.